File: api/src/test/java/net/blueshell/api/factory/model/EventBannerFactory.java
package net.blueshell.api.factory.model;

import net.blueshell.api.testutil.ModelTestUtils;
import com.github.javafaker.Faker;
import lombok.RequiredArgsConstructor;
import net.blueshell.api.model.File;
import net.blueshell.api.model.event.Event;
import net.blueshell.api.model.event.EventBanner;
import org.springframework.stereotype.Component;

import java.util.concurrent.atomic.AtomicLong;

/**
 * Factory for EventBanner model test instances.
 */
@Component
@RequiredArgsConstructor
public class EventBannerFactory {

    private static final AtomicLong counter = new AtomicLong(1000);
    private final Faker faker;
    private final EventFactory eventFactory;
    private final FileFactory fileFactory;

    public EventBanner createBasic() {
        EventBanner eb = new EventBanner();

        Event event = eventFactory.createBasic();
        File file = fileFactory.createImage();

        eb.setEvent(event);
        eb.setFile(file);

        return eb;
    }

    public EventBanner createFull() {
        return createBasic();
    }

    public EventBanner createWithCustomizations(java.util.function.Consumer<EventBanner> customizer) {
        EventBanner eb = createFull();
        customizer.accept(eb);
        return eb;
    }

    private Long generateId() {
        return counter.incrementAndGet();
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/model/event/EventBannerFactory.kt
package net.blueshell.api.factory.model.event

import com.github.javafaker.Faker
import net.blueshell.api.domain.event.persistence.EventBanner
import net.blueshell.api.factory.model.FileFactory
import org.springframework.stereotype.Component
import java.util.concurrent.atomic.AtomicLong
import java.util.function.Consumer

/**
 * Factory for EventBanner model test instances.
 */
@Component
class EventBannerFactory(
    private val faker: Faker,
    private val eventFactory: EventFactory,
    private val fileFactory: FileFactory
) {

    fun createBasic(): EventBanner {
        val banner = EventBanner()
        val event = eventFactory.createBasic()
        val file = fileFactory.createImage()
        banner.event = event
        banner.id.fileId = file.id
        return banner
    }

    fun createFull(): EventBanner = createBasic()

    fun createWithCustomizations(customizer: Consumer<EventBanner>): EventBanner {
        val banner = createFull()
        customizer.accept(banner)
        return banner
    }

    private fun generateId(): Long = COUNTER.incrementAndGet()

    private companion object {
        val COUNTER = AtomicLong(1000)
    }
}
